alica_engine/src/engine/modelmanagement/ModelManager.cpp

```diff
@@ -36,13 +40,22 @@ ModelManager::ModelManager(ConfigChangeListener& configChangeListener, const std
     Factory::setModelManager(this);
 }
 
-Plan* ModelManager::loadPlanTree(const std::string& masterPlanName)
+Plan* ModelManager::loadPlanTree(const std::string& masterPlanName, std::optional<std::string> placeholderMapping)
 {
     std::string masterPlanPath;
     if (masterPlanPath = findFile(masterPlanName + alica::Strings::plan_extension); masterPlanPath.empty()) {
         AlicaEngine::abort(LOGNAME, "Cannot find MasterPlan '", masterPlanName, "'");
     }
 
+    std::string placeholderMappingPath;
+    if (placeholderMapping.has_value()) {
```

Comment:
empty filename is a valid name?

Reply:
this is changed to be a json now instead of a file name, therefore I think have an optional is useful